src/service.rs

```diff
@@ -300,7 +498,15 @@ impl Service {
 
     /// The main execution loop of the discv5 serviced.
     async fn start(&mut self) {
-        tracing::info!("{:?}", self.config.ip_mode);
+        // In the case where not many peers populate the topic's kbuckets, ensure topics keep being republished.
+        let mut registration_interval = tokio::time::interval(REGISTER_INTERVAL);
+        let mut topics_to_reg_iter = self
+            .registration_attempts
+            .keys()
+            .map(|topic| (topic.clone(), topic.hash()))
+            .collect::<Vec<(Topic, TopicHash)>>()
+            .into_iter();
+
```

Comment:
this means topics to register are locked to whatever is in the beginning

Reply:
This is the idea to make sure all topics are registered eventually and to avoid blocking Service for too long with registering topics